tests/key_field/key_field_sample_generated.h

```diff
@@ -170,13 +200,376 @@ inline bool operator!=(const Bar &lhs, const Bar &rhs) {
 }
 
 
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) Color FLATBUFFERS_FINAL_CLASS {
+ private:
+  float rgb_[3];
+  uint8_t tag_;
+  int8_t padding0__;  int16_t padding1__;
+
+ public:
+  static const flatbuffers::TypeTable *MiniReflectTypeTable() {
+    return ColorTypeTable();
+  }
+  Color()
+      : rgb_(),
+        tag_(0),
+        padding0__(0),
+        padding1__(0) {
+    (void)padding0__;
+    (void)padding1__;
+  }
+  Color(uint8_t _tag)
+      : rgb_(),
+        tag_(flatbuffers::EndianScalar(_tag)),
+        padding0__(0),
+        padding1__(0) {
+    (void)padding0__;
+    (void)padding1__;
+  }
+  Color(flatbuffers::span<const float, 3> _rgb, uint8_t _tag)
+      : tag_(flatbuffers::EndianScalar(_tag)),
+        padding0__(0),
+        padding1__(0) {
+    flatbuffers::CastToArray(rgb_).CopyFromSpan(_rgb);
+    (void)padding0__;
+    (void)padding1__;
+  }
+  const flatbuffers::Array<float, 3> *rgb() const {
+    return &flatbuffers::CastToArray(rgb_);
+  }
+  flatbuffers::Array<float, 3> *mutable_rgb() {
+    return &flatbuffers::CastToArray(rgb_);
+  }
+  bool KeyCompareLessThan(const Color * const o) const {
+    return KeyCompareWithValue(o->rgb()) < 0;
+  }
+  int KeyCompareWithValue(const flatbuffers::Array<float, 3> *_rgb) const {
+    const flatbuffers::Array<float, 3> *curr_rgb = rgb();
+    for (flatbuffers::uoffset_t i = 0; i < curr_rgb->size(); i++) {
+      const auto lhs = curr_rgb->Get(i);
+      const auto rhs = _rgb->Get(i);
+      if(lhs != rhs)
+        return static_cast<int>(lhs > rhs) - static_cast<int>(lhs < rhs);
+    }
+    return 0;
+  }
+  uint8_t tag() const {
+    return flatbuffers::EndianScalar(tag_);
+  }
+  void mutate_tag(uint8_t _tag) {
+    flatbuffers::WriteScalar(&tag_, _tag);
+  }
+};
+FLATBUFFERS_STRUCT_END(Color, 16);
+
+inline bool operator==(const Color &lhs, const Color &rhs) {
+  return
+      (lhs.rgb() == rhs.rgb()) &&
```

Comment:
As return value of `rgb `function is `const flatbuffers::Array<float, 3> *` which is a pointer, IMO this compare is not currect. Maybe flatbuffers::Array operator == needed to be implemented.

Reply:
Let me create another issue for this.

Reply:
Issue created #7746 , I'll address that later.